lib/datadog_backup/client.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+require 'faraday'
+require 'faraday/retry'
+
+module DatadogBackup
+  class Client
+    RETRY_OPTIONS = {
+      max: 5,
+      interval: 0.05,
+      interval_randomness: 0.5,
+      backoff_factor: 2
+    }.freeze
+
+    def initialize
+      @client = Faraday.new(
+        url: ENV.fetch('DD_SITE_URL', 'https://api.datadoghq.com/'),
```

Comment:
The default url might be better stored as a class constant, eg:

    DEFAULT_DD_SITE_URL = 'https://api.datadoghq.com/'

And later:

    url: ENV.fetch('DD_SITE_URL', DEFAULT_DD_SITE_URL)